EventListener/ExceptionListener.php

```diff
@@ -34,15 +35,25 @@ class ExceptionListener implements EventSubscriberInterface
 {
     private $exceptionListener;
     private $dispatcher;
+    private $innerExceptionListener;
 
-    public function __construct($controller, ?LoggerInterface $logger, EventDispatcherInterface $dispatcher)
+    public function __construct($controller, ?LoggerInterface $logger, $debug = false, EventDispatcherInterface $dispatcher, $innerExceptionListener = null)
     {
         if (class_exists(ErrorListener::class)) {
-            $this->exceptionListener = new ErrorListener($controller, $logger);
+            $this->exceptionListener = new ErrorListener($controller, $logger, $debug);
```

Comment:
I am not sure that we really need the `$exceptionListener` property anymore now. If I don't miss anything we would use the decorated exception listener instead.

Reply:
it is still needed. see the last line of `\FOS\RestBundle\EventListener\ExceptionListener::onKernelException()`, we still let the core listener handle the actual response creation.

Reply:
Don't we have to delegate that to the decorated exception listener now? Something like this:

```php
public function __construct($controller, ?LoggerInterface $logger, EventDispatcherInterface $dispatcher, $decoratedExceptionListener = null)
{
    if (null !== $decoratedExceptionListener) {
        $this->exceptionListener = $decoratedExceptionListener;
    } elseif (class_exists(ErrorListener::class)) {
        $this->exceptionListener = new ErrorListener($controller, $logger);
    } else {
        $this->exceptionListener = new LegacyExceptionListener($controller, $logger);
    }

    $this->dispatcher = $dispatcher;
}
```

Reply:
basically the logic I implemented checks the zone config and if its within the zone .. it uses the `exceptionListener` and if its outside the zone it uses `innerExceptionListener`